sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/GlobalThroughputControlConfigBuilder.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Throughput global control config builder.
+ */
+@Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+public class GlobalThroughputControlConfigBuilder {
+    private final CosmosAsyncContainer controlContainer;
+
+    private Duration controlItemRenewInterval;
+    private Duration controlItemExpireInterval;
+
+    GlobalThroughputControlConfigBuilder(CosmosAsyncClient client, String databaseId, String containerId) {
+        checkNotNull(client, "Client can not be null");
+        checkArgument(StringUtils.isNotEmpty(databaseId), "DatabaseId cannot be null nor empty");
+        checkArgument(StringUtils.isNotEmpty(containerId), "ContainerId cannot be null nor empty");
+
+        controlContainer = client.getDatabase(databaseId).getContainer(containerId);
+    }
+
+    /**
+     * Set the control item renew interval.
+     *
+     * This controls how often the client is going to update the throughput usage of itself
+     * and adjust its own throughput share based on the throughput usage of other clients.
+     *
+     * In short words, it controls how quickly the shared throughput will reload balanced across different clients.
+     *
+     * @param controlItemRenewInterval The control item renewal interval.
+     * @return The {@link GlobalThroughputControlConfigBuilder}
+     */
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public GlobalThroughputControlConfigBuilder setControlItemRenewInterval(Duration controlItemRenewInterval) {
+        this.controlItemRenewInterval = controlItemRenewInterval;
+        return this;
+    }
+
+    /**
+     * Set the control item expire interval.
+     *
+     * A client may be offline due to various reasons (being shutdown, network issue... ).
+     * This controls how quickly we will detect the client has been offline and hence allow its throughput share to be taken by other clients.
+     *
+     * @param controlItemExpireInterval The control item expire interval.
+     * @return The {@link GlobalThroughputControlConfigBuilder}
+     */
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public GlobalThroughputControlConfigBuilder setControlItemExpireInterval(Duration controlItemExpireInterval) {
+        this.controlItemExpireInterval = controlItemExpireInterval;
+        return this;
+    }
+
+    /**
+     * Validate the throughput global control configuration and create a new throughput global control config item.
+     *
+     * @return A new {@link GlobalThroughputControlConfig}.
+     */
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public GlobalThroughputControlConfig build() {
+        if (this.controlItemExpireInterval != null && this.controlItemRenewInterval != null) {
+            if (this.controlItemExpireInterval.getSeconds() < this.controlItemRenewInterval.getSeconds()) {
```

Comment:
getControlItemExpireInterval controls the ttl of the document associated with the client
getControlItemRenewInterval controls how often we update it.

let's say
controlItemExpireInterval = controlItemRenewInterval
there might be some scheduling shift, etc which may cause the document gets ttl-ed and then we attempt to update again around the same time.

I think you need to be more strict and at least attempt to update the document twice in its ttl threshold to ensure it will not get purged.

so instead of throw if controlItemExpireInterval < controlItemRenewInterval
you should
 always at least we should have 2 * controlItemRenewInterval + 1 < controlItemExpireInterval to proceed.

Reply:
@xinlian12  @FabianMeiswinkel thought?

Reply:
Make sense. And also I think can use this to answer the following question about what if one of them is null.
By using  2 * controlItemRenewInterval + 1 = controlItemExpireInterval to calculate the other

Reply:
Updated to the suggestion calculation.